### ISSUE TITLE
Expose LDProvider as a standalone export

### ISSUE THREAD

**Issue by nimi** [launchdarkly/react-client-sdk#27]
**Is your feature request related to a problem? Please describe.**

There are two documented ways to initialize the React SDK -- by using the the synchronous HOC `withLDProvider` or using an async function, `asyncWithLDProvider`, which is designed to resolve the javascript client and flags via HTTP before doing an initial render. Both of these solutions are problematic for our use case. The list below describes the issues we're facing

- First, `asyncWithLDProvider` requires we block rendering of our app before we have access to the Provider. This requires us to write custom logic outside of the react app to display a loading state. This is handled in our React app, so we now have to manage two different places where loading states are enabled/disabled
- Second, both `asyncWithLDProvider` and `withLDProvider`, we're unable to feed dynamic state into the initialization. We use redux to store global configuration and user state based on environment and auth context. As a HOC, we're unable to initialize our client with the correct clientID and user state.

A related issue is filed here https://github.com/launchdarkly/js-client-sdk/issues/174

**Describe the solution you'd like**

I'd like to suggest, that exposing the `LDProvider` as a standalone component which can accept the provider configuration. Something like:

```typescript
return (
  <LDProvider {...myLDPropsFromRedux} />
   <App />
  </LDProvider>
)
```

Would this be something you be open as a PR submission or have further thoughts on?

**Describe alternatives you've considered**
Using the js SDK instead

**Comment by yusinto** [comment]
@nimi thank you for raising this. We are currently reviewing your pr #28 and will have some updates soon.

**Comment by vlavellauruit** [comment]
Any updates on this? We also really need this feature

**Comment by eli-darkly** [comment]
We reviewed the corresponding pull request and made some requests for changes, so we were still waiting for @nimi to follow up. If we can't get in touch with the contributor we'll probably proceed with a similar change on our end soon.

**Comment by goldenshun** [comment]
Just want to point out this is also an issue because the HOC does not hoist React statics. We have multiple applications that use Next.js's `getInitialProps`, and they all will break if we try to use the `withLDProvider` HOC.

**Comment by KiranGopalakrishnan** [comment]
we kind of ran into a similar problem , I worked around it by creating a component that received the user data from the store and initialize LD using `withLDProvider` and do something like 
`
const WithFeatureFlags: React.FC<Props> = ({
    user
}: Props) => {
    if(user){
        const Child = () => <>{children}</>;

        const WithLDContext = withLDProvider({
            clientSideID: launchDarklyClientId,
            user,
            reactOptions: {
                useCamelCaseFlagKeys: false,
            },
        })(Child);

        return <WithLDContext />;
    }

    return <>{children}</>;
};
`

And wrap this around my App at the root level

**Comment by bwoskow-ld** [comment]
We have merged #31 to our `contrib` branch and will include this in the next version of our React SDK, slated for release in the next week or two. Stay tuned.

**Comment by bwoskow-ld** [comment]
This is released in [2.19.0](https://github.com/launchdarkly/react-client-sdk/releases/tag/2.19.0).

**Comment by nimi** [comment]
Hey y'all! Sorry I wasn't able to get back on this one to finish out. Pandemic + work + kiddo at home didn't leave me much time for much else. Thank you so much for getting this one into the release, it looks ✨ . We look forward to getting this into our app 😃
